src/gui/qgsadvanceddigitizingdockwidget.cpp

```diff
@@ -144,6 +149,29 @@ QgsAdvancedDigitizingDockWidget::QgsAdvancedDigitizingDockWidget( QgsMapCanvas *
   {
     commonAngles << QPair<double, QString>( *it, formatCommonAngleSnapping( *it ) );
   }
+
+  {
+    QMenu *snappingPriorityMenu = new QMenu( tr( "Snapping Priority" ), mCommonAngleActionsMenu );
+    QActionGroup *snappingPriorityActionGroup = new QActionGroup( snappingPriorityMenu );
+    QAction *featuresAction = new QAction( tr( "Prioritise Snapping to Features" ), snappingPriorityActionGroup );
+    featuresAction->setCheckable( true );
+    QAction *anglesAction = new QAction( tr( "Prioritise Snapping to Angle" ), snappingPriorityActionGroup );
```

Comment:
I'd suggest to make sure it's not confused with proper angle snapping:
```suggestion
    QAction *anglesAction = new QAction( tr( "Prioritise Snapping to Common Angle" ), snappingPriorityActionGroup );
```